### PR TITLE
chore: improve ruff coverage

### PR DESCRIPTION
**Pull request by daniel-makerx**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -32,34 +32,79 @@ poethepoet = "^0.19.0"
...
-    "buck-out",
     "build",
     "dist",
     "node_modules",
     "venv",
+    "docs/sphinx",
 ]
-per-file-ignores = {}
 # Allow unused variables when underscore-prefixed.
 dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
```
Comment (achidlow) [inline pyproject.toml:109]: whilst you're here - this one can actually be omitted, this is the default :D

File: pyproject.toml [hunk 2]
```diff
@@ -32,34 +32,79 @@ poethepoet = "^0.19.0"
...
-    ".tox",
     ".venv",
     "__pypackages__",
     "_build",
-    "buck-out",
     "build",
     "dist",
     "node_modules",
     "venv",
+    "docs/sphinx",
```
Comment (achidlow) [inline pyproject.toml:106]: also, we can omit all of the above, and just have

```toml
extend-exclude = [
  "docs/"
]
```

File: src/algokit_utils/application_specification.py [hunk 3]
```diff
@@ -62,7 +63,7 @@ class DefaultArgumentDict(TypedDict):
     data: int | str | bytes | MethodDict
 
 
-StateDict = TypedDict(  # noqa: UP013  # can't convert as "global" is a reserved keyword
+StateDict = TypedDict(  # can't convert as "global" is a reserved keyword
```
Comment (achidlow) [inline src/algokit_utils/application_specification.py:66]: oh where did the `noqa` go? did ruff get a smarts boost?

Comment (daniel-makerx) [reply src/algokit_utils/application_specification.py:66]: Yeah latest version said it wasn't required anymore and removed it

Comment (daniel-makerx) [reply src/algokit_utils/application_specification.py:66]: Just occurred to me that comment that is left is not needed now

File: pyproject.toml (hunk 2 above)
Comment (daniel-makerx) [reply pyproject.toml]: fixed

File: pyproject.toml (hunk 1 above)
Comment (daniel-makerx) [reply pyproject.toml]: done

File: src/algokit_utils/application_specification.py (hunk 3 above)
Comment (daniel-makerx) [inline src/algokit_utils/application_specification.py:66]:
```suggestion
StateDict = TypedDict(
```

Comment (daniel-makerx) [reply src/algokit_utils/application_specification.py]: Removed comment

Comment (achidlow) [reply src/algokit_utils/application_specification.py]: maybe `need to use function-form of TypedDict here since "global" is a reserved keyword`?

**Review by achidlow — APPROVED** [review]

**Pull request merged**